src/config.c

```diff
@@ -241,7 +241,7 @@ void cfg_init(char *name, int allow_missing)
 		file = fopen(cfg_name, "r");
 		if (!file) {
 			if (allow_missing && errno == ENOENT) return;
-			pexit("fopen: %s", cfg_name);
+			pexit("fopen: %s", path_expand(cfg_name));
```

Comment:
It's already this way in core, but shouldn't jumbo's `cfg_name = str_alloc_copy(path_expand_ex(name));` have resulted in `cfg_name` containing an already expanded path? If it doesn't, then maybe that's the problem?

Also, core tries not to modify the global `cfg_name` until `cfg_init` succeeds and is about to return. Jumbo may modify `cfg_name` yet hit the semi-error return on `if (allow_missing && errno == ENOENT) return;` - I wonder if this has undesirable side-effects. I guess whoever made the changes to `cfg_init` in jumbo didn't think of that.

For context, here's what it is in core:

```c
void cfg_init(const char *name, int allow_missing)
{
        FILE *file;
        char line[LINE_BUFFER_SIZE];
        int number;

        if (cfg_database) return;

        if (!(file = fopen(path_expand(name), "r"))) {
                if (allow_missing && errno == ENOENT) return;
                pexit("fopen: %s", path_expand(name));
        }

        number = 0;
        while (fgetl(line, sizeof(line), file))
        if (cfg_process_line(line, ++number)) cfg_error(name, number);

        if (ferror(file)) pexit("fgets");

        if (fclose(file)) pexit("fclose");

        cfg_name = str_alloc_copy(path_expand(name));
}
```

Reply:
Introduced by de989e6e12f48090f1b3dd095ea327ed8a2df30e

> `cfg_name = str_alloc_copy(path_expand_ex(name));` have resulted in cfg_name containing an already expanded path?

No

> If it doesn't, then maybe that's the problem?

Not in my opinion. `path_expand_ex()` handle `--config=file_name`; it was empty in the example.